### ISSUE TITLE
GDAL Processings crash when called from Python API

### ISSUE THREAD

**Issue by Kanahiro** [qgis/QGIS#45766]
### What is the bug or the crash?

Any GDAL Processings crash when called from Python API.

### Steps to reproduce the issue

GDAL Processing can be called from GUI, it works.

![スクリーンショット 2021-11-01 16 30 26](https://user-images.githubusercontent.com/20744195/139636861-3123cdee-d34f-4a2b-9fc4-36bc3be0dcfe.png)

However, when called from Python API like following, <b>QGIS will crash.</b>

```python
import processing
processing.run("gdal:slope", params)
```

### Versions

QGISバージョン | 3.22.0-Białowieża | QGISコードリビジョン | d9022691f1
-- | -- | -- | --
Qtバージョン | 5.15.2
Pythonバージョン | 3.9.5
GDAL/OGRバージョン | 3.3.2
PROJバージョン | 8.1.1
EPSG Registry databaseバージョン | v10.028 (2021-07-07)
GEOSバージョン | 3.10.0-CAPI-1.16.0
SQLiteバージョン | 3.35.2
PDALバージョン | 2.3.0
PostgreSQLクライアントバージョン | 13.0
SpatiaLiteバージョン | 5.0.1
QWTバージョン | 6.1.3
QScintilla2バージョン | 2.11.5
OSバージョン | Windows 10 Version 1909

### Supported QGIS version

- [x] I'm running a supported QGIS version according to the roadmap.

### New profile

- [x] I tried with a new QGIS profile

**Comment by Kanahiro** [comment]
GDAL Python-binding works.

```python
# works
from osgeo import gdal
gdal.Warp(dst, src, **params)
```

```python
# DOESN'T work
import processing
processing.run("gdal:warpreproject", params)
```

**Comment by nicogodet** [comment]
Crash ?
Any stack trace ?

Looks similar but different to this issue https://github.com/qgis/QGIS/issues/45685

**Comment by gioman** [comment]
> Looks similar but different to this issue #45685

@nicogodet seems related

@Kanahiro works fine on 3.16?

**Comment by Kanahiro** [comment]
sorry, _crash_ is not a correct word, _freeze_ is more appropriate.

**Comment by nicogodet** [comment]
@gioman Yeah it seems except running gdal algorithm through processing python API was working fine when I tried the model converted to python script.

@Kanahiro Does it crash exclusively when called in Python console ?

**Comment by nicogodet** [comment]
And other than gdal alg crash too or just warp ?

**Comment by Kanahiro** [comment]
> Does it crash exclusively when called in Python console ?

occurs also in plugin scripts.

I tried two GDAL processing, `gdal:warpreproject` and `gdal:slope`.
These also cause freeze.

`qgis:slope` works well.

**Comment by nicogodet** [comment]
@gioman Yeah it works fine on 3.16.12

**Comment by gioman** [comment]
> @gioman Yeah it works fine on 3.16.12

@nicogodet added the regression label.

**Comment by nyalldawson** [comment]
see https://github.com/qgis/QGIS/pull/45128#issuecomment-958529173 -- caused by the update to the sip library

**Comment by nicogodet** [comment]
Fixed in OSGeo4W v2 3.22.0-4 and 3.16.12-4
